### PR TITLE
Enable to set max_key_fee to null via cli

### PR DESCRIPTION
**Pull request by mirgee**
This fix is meant to address #2050. Instead letting argparse check the number of arguments, the arguments are cast to a list whose length is checked in `_parse_list()`. It was tested via both CLI and yaml, and I extended some of the unit tests in `test_conf.py`.

### CODE REVIEW

**Comment by tzarebczan** [comment]
Thanks for the PR! We'll get a review on this soon. Can we show you some [appreciation](https://lbry.com/faq/appreciation) for the contribution?

File: lbry/lbry/conf.py [hunk 1]
```diff
@@ -148,7 +148,10 @@ def validate(self, value):
 
     @staticmethod
     def _parse_list(l):
-        assert len(l) == 2, 'Max key fee is made up of two values: "AMOUNT CURRENCY".'
+        if len(l) == 1 and (l[0] == 'null' or not l[0]):
+            return None
+        assert len(l) == 2, ('Max key fee is made up of either two values: '
+                             '"AMOUNT CURRENCY", or "null" (to set no limit)')
```
Comment (eukreign) [inline lbry/lbry/conf.py]:
```
        assert len(l) == 2, \
            'Max key fee is made up of either two values: "AMOUNT CURRENCY", or "null" (to set no limit)'
```
or
```
        assert len(l) == 2, (
            'Max key fee is made up of either two values: '
            '"AMOUNT CURRENCY", or "null" (to set no limit)'
        )
```
is the format we normally use.

File: lbry/lbry/conf.py [hunk 2]
```diff
@@ -159,8 +162,8 @@ def _parse_list(l):
         return {'amount': amount, 'currency': currency}
 
     def deserialize(self, value):
-        if value is None:
-            return
+        if not value:
+            return None
```
Comment (eukreign) [inline lbry/lbry/conf.py]: Why this change?

File: lbry/tests/unit/test_conf.py [hunk 3]
```diff
@@ -222,6 +227,14 @@ def test_max_key_fee_from_args(self):
         c = Config.create_from_arguments(args)
         self.assertEqual(c.max_key_fee, None)
 
+        args = parser.parse_args(['--max-key-fee', 'null'])
+        c = Config.create_from_arguments(args)
+        self.assertEqual(c.max_key_fee, None)
+
+        args = parser.parse_args(['--max-key-fee', ''])
+        c = Config.create_from_arguments(args)
+        self.assertEqual(c.max_key_fee, None)
```
Comment (eukreign) [inline lbry/tests/unit/test_conf.py]: I think `null` as an argument is enough, supporting a blank is ambiguous and might have unintended edge cases.

File: lbry/lbry/conf.py (hunk 2 above)
Comment (mirgee) [reply lbry/lbry/conf.py]: To support the empty string case, which we don't want to do, see below; I'll revert it.

File: lbry/lbry/conf.py [hunk 4]
```diff
@@ -148,7 +148,12 @@ def validate(self, value):
 
     @staticmethod
     def _parse_list(l):
-        assert len(l) == 2, 'Max key fee is made up of two values: "AMOUNT CURRENCY".'
+        if len(l) == 1 and l[0] == 'null':
```
Comment (eukreign) [inline lbry/lbry/conf.py]: I think this could be simplified to:
```
if l == ['null']:
```

File: lbry/tests/unit/test_conf.py [hunk 5]
```diff
@@ -207,6 +207,11 @@ def test_max_key_fee_from_yaml(self):
                 c.max_key_fee = None
             with open(config, 'r') as fd:
                 self.assertEqual(fd.read(), 'max_key_fee: null\n')
+            c = Config.create_from_arguments(
+                types.SimpleNamespace(config=config)
+            )
+            with open(config, 'r') as fd:
+                self.assertEqual(c.max_key_fee, None)
```
Comment (eukreign) [inline lbry/tests/unit/test_conf.py]: Can you explain what's going on here? I don't understand what this is testing. (why the new `c` config instance and why the assert for None, also why that assert happens in a file context manager but the `fd` descriptor isn't used)

Comment (mirgee) [reply lbry/tests/unit/test_conf.py]: This is trying to test `max_key_fee` is set to `None` that when config is created from `settings.yaml` where `max_key_fee: null\n` (the context manager shouldn't be there). It doesn't test my changes (or our code, for that matter), so I will remove it.
